### ISSUE TITLE
Can't insert a nested document if there is a key dependency to it

### ISSUE THREAD

**Issue by matko** [terminusdb/terminusdb#1718]
When a document has a key dependency to a document field, it is impossible to insert both at once.

Example, given the following schema:
```
{ "@type": "Foo",
  "@key": {"@type":"Lexical", "@fields":["bar"]},
  "bar": "Bar"
}
{ "@type":"Bar",
  "x": "xsd:string"
} 
```

Inserting this document:
```
{"bar": {"x":a"}}
```
Will give an ugly stack trace because the toplevel is unable to figure out what the id is for the nested document, and therefore cannot generate its id.

This should actually be possible to do by first generating the id for the nested document. We normally don't do that cause nested documents can be subdocuments, whose ID depends on its parent, but this is not the case for ordinary documents.

**Comment by matko** [comment]
Will be solved in #1719

**Comment by matko** [comment]
#1719 solved this, so closing.
